### PR TITLE
Rjd 1278/traffic simulator update

### PR DESCRIPTION
**Pull request by robomic**
# Description

## Abstract

This PR contains revisions of the traffic_simulator package unit tests.

## Details
Changes are minimal, their main purpose is to unify the source code with previously adopted rules.
Tests have been sorted.
Tests that have been refactored, now contain descriptions.

List of tests updated in this PR:
1. helper.constructActionStatus
2. helper.constructLaneletPose
3. helper.constructRPY
4. helper.constructRPYfromQuaternion
5. helper.constructPose
6. helper.constructLidarConfiguration
7. helper.constructDetectionSensorConfiguration
8. SimulationClock.SimulationClock
9. TrafficLightManager.getTrafficLight
10. TrafficLightManager.getTrafficLights_setColor
11. TrafficLightManager.getTrafficLights_setArrow
12. Color.Color
13. Color.make
14. Color.make_wrong
15. Status.Status
16. Status.make
17. Status.make_wrong
18. Shape.Shape
19. Shape.make
20. Shape.make_wrong
21. Bulb.Bulb
22. Bulb.make
23. Bulb.make_wrong
24. Bulb.operator_TrafficLight
25. TrafficLight.contains_colorStatusShape
26. TrafficLight.set_valid

## References
Jira ticket: [internal link](https://tier4.atlassian.net/browse/RJD-1278)

# Destructive Changes

There are no destructive changes.

### CODE REVIEW

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp [hunk 1]
```diff
@@ -723,39 +726,68 @@ TEST(TrafficLight, Bulb_trafficLightMessageConversion)
...
  */
-TEST(TrafficLight, Shape_make_wrong)
+TEST_F(TrafficLightTest, contains_colorStatusShape)
 {
-  EXPECT_THROW(traffic_simulator::TrafficLight::Shape::make("wrong_shape"), common::SyntaxError);
-}
+  using TrafficLight = traffic_simulator::TrafficLight;
+  using Color = TrafficLight::Color;
+  using Status = TrafficLight::Status;
+  using Shape = TrafficLight::Shape;
```
Comment (TauTheLepton) [inline simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp:753]: I think this should be moved to the fixture class

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp [hunk 2]
```diff
@@ -723,39 +726,68 @@ TEST(TrafficLight, Bulb_trafficLightMessageConversion)
...
+ * @note Test function behavior with a valid string.
  */
-TEST(TrafficLight, Bulb_make_wrong)
+TEST_F(TrafficLightTest, set_valid)
 {
-  using Bulb = traffic_simulator::TrafficLight::Bulb;
+  using TrafficLight = traffic_simulator::TrafficLight;
+  using Color = TrafficLight::Color;
+  using Status = TrafficLight::Status;
+  using Shape = TrafficLight::Shape;
```
Comment (TauTheLepton) [inline simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp:785]: I think this should be moved to the fixture class

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp [hunk 3]
```diff
@@ -18,37 +18,49 @@
...
-  std::string path =
-    ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm";
-  geographic_msgs::msg::GeoPoint origin;
-  origin.latitude = 35.61836750154;
-  origin.longitude = 139.78066608243;
-  const auto hdmap_utils_ptr = std::make_shared<hdmap_utils::HdMapUtils>(path, origin);
-  traffic_simulator::TrafficLightManager manager(hdmap_utils_ptr);
+  using Color = traffic_simulator::TrafficLight::Color;
+  using Status = traffic_simulator::TrafficLight::Status;
+  using Shape = traffic_simulator::TrafficLight::Shape;
```
Comment (TauTheLepton) [inline simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp:62]: I think this should be moved to the fixture class

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp [hunk 4]
```diff
@@ -63,20 +75,15 @@ TEST(TrafficLightManager, setColor)
...
```
Comment (TauTheLepton) [inline simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp:85]: I think this should be moved to the fixture class

**Review by TauTheLepton — COMMENTED** [review]
Please see my comments

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp (hunk 3 above)
Comment (robomic) [reply simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp:62]: I will just move usings to the top of the file

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp (hunk 4 above)
Comment (robomic) [reply simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light_manager.cpp:85]: I will just move usings to the top of the file

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp (hunk 1 above)
Comment (robomic) [reply simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp:753]: I will just move usings to the top of the file

File: simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp (hunk 2 above)
Comment (robomic) [reply simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp:785]: I will just move usings to the top of the file

**Pull request merged**